pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -177,20 +187,40 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	// Compare the desired and observed resources to determine our situation.
 	// We fetch private endpoints here, since for scaling we're interested in the actual
 	// state of the deployment.
-	got := 0
+	ready, notReady := 0, 0
 
 	// Propagate service name.
 	pa.Status.ServiceName = sks.Status.ServiceName
 	// Currently, SKS.IsReady==True when revision has >0 ready pods.
 	if sks.Status.IsReady() {
-		podCounter := resourceutil.NewScopedEndpointsCounter(c.endpointsLister, pa.Namespace, sks.Status.PrivateServiceName)
-		got, err = podCounter.ReadyCount()
+		podEndpointCounter := resourceutil.NewScopedEndpointsCounter(c.endpointsLister, pa.Namespace, sks.Status.PrivateServiceName)
+		ready, err = podEndpointCounter.ReadyCount()
+		if err != nil {
+			return fmt.Errorf("error checking endpoints %s: %w", sks.Status.PrivateServiceName, err)
+		}
+
+		notReady, err = podEndpointCounter.NotReadyCount()
 		if err != nil {
 			return fmt.Errorf("error checking endpoints %s: %w", sks.Status.PrivateServiceName, err)
 		}
 	}
-	logger.Infof("PA scale got=%d, want=%d, ebc=%d", got, want, decider.Status.ExcessBurstCapacity)
-	return computeStatus(pa, want, got)
+
+	podCounter := resourceutil.NewNotRunningPodsCounter(c.podsLister, pa.Namespace, sks.Status.ServiceName)
```

Comment:
I have some level of grief with us passing `sks.Status.ServiceName` in this case. That doesn't reflect what we're querying in the counter at all, it just happens to be named like the revision itself. We should instead either:

1. Query the revision name from the PAs labels
2. Pass the PA Name
3. Fetch the ScaleTargetRef from the PA spec and use that to determine what to query

Reply:
went with option 1.